File: java/unidad_02/Problema_2_08/src/Problema_2_08.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dado como dato el monto de la compra de un cliente, determina lo
 * que el mismo debe pagar teniendo en cuenta una serie de descuentos.
 */

public class Problema_2_08 {

    public static float[] aplicarDescuento(float totalCompra) {

        float descuento, nuevoTotal;

        if (totalCompra < 500) {

            descuento = 0;

        } else if (totalCompra <= 1000) {

            descuento = (float) 0.05;

        } else if (totalCompra <= 7000) {

            descuento = (float) 0.11;

        } else if (totalCompra <= 15000) {

            descuento = (float) 0.18;

        } else {

            descuento = (float) 0.25;
        }
        nuevoTotal = totalCompra - totalCompra * descuento;

        return new float[] { descuento, nuevoTotal };
    }

    public static void main(String[] args) {

        float totalCompra;

        float[] valores;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Total de compra: ");
            totalCompra = scanner.nextFloat();

            if (totalCompra <= 0) {

                System.out.println("Error: el total de compra debe ser mayor que cero.");
                return;
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        valores = aplicarDescuento(totalCompra);

        System.out.printf("%.2f u tiene %.0f %% de descuento, %n", totalCompra, valores[0] * 100);

        System.out.printf("por lo tanto, el nuevo total es de %.2f u", valores[1]);
    }
}

File: python/unidad_02/problema_2_08.py
"""
El programa, dado como dato el monto de la compra de un cliente, determina lo
que el mismo debe pagar teniendo en cuenta una serie de descuentos.
"""

def aplicar_descuento(total_compra):

    if total_compra < 500:

        descuento = 0

    elif total_compra <= 1000:

        descuento = 0.05

    elif total_compra <= 7000:

        descuento = 0.11

    elif total_compra <= 15000:

        descuento = 0.18

    else:

        descuento = 0.25
        
    nuevo_total = total_compra - total_compra * descuento
    
    return descuento, nuevo_total

def main():

    try:

        total_compra = float(input("Total de compra: "))

        if total_compra <= 0:

            print("Error: el total de compra debe ser mayor que cero.")
            return

    except ValueError as e:

        print(e)
        return
    
    descuento, nuevo_total = aplicar_descuento(total_compra)
    
    print(f"{total_compra} u tiene {descuento * 100} % de descuento,")

    print(f"por lo tanto, el nuevo total es de {nuevo_total:2f} u")

if __name__ == '__main__':

    main()